### PR TITLE
Implements basic tests for database information retrieval

### PR DESCRIPTION
**Pull request by matt-wigg**
All tests are passing with correct assertion counts and values

Trello Ticket: [LINK1](https://trello.com/c/Dk2fEJOS/18-write-a-test-to-check-the-seed-information-is-going-into-my-database-correctly) [LINK2](https://trello.com/c/2ws41pjg/24-research-how-jest-and-puppeteer-work-to-write-tests)

### CODE REVIEW

File: test/test.js [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+  test('Should get review comments for all product id', async (done) => {
+    let id = 0;
+    const queryString = 'SELECT * FROM user_reviews WHERE url_id = ?';
+    const promises = [];
+    while (id < 100) {
+      db.query(queryString, [id], (err, results) => {
+        promises.push(results);
+        if (promises.length === 100) {
+          for (let i = 0; i < promises.length; i += 1) {
+            expect(promises[i][0].comment).toBeDefined();
```
Comment (andrewdeanyoder) [inline test/test.js]: Hi @Matt-Wigg, check out the answer to this question: https://github.com/facebook/jest/issues/3457 Maybe you could just use "typeof" to test whether the comment is a string?

File: test/test.js [hunk 2]
```diff
@@ -36,4 +37,96 @@ describe('Database interactions', () => {
...
     }
   });
+  test('Should get names for all product id entries', (done) => {
+    let id = 0;
+    const queryString = 'SELECT * FROM user_reviews WHERE url_id = ?';
+    const promises = [];
+    while (id < 100) {
+      expect.assertions(100);
+      db.query(queryString, [id], (err, results) => {
+        promises.push(results);
```
Comment (andrewdeanyoder) [inline test/test.js:47]: Shouldn't there be some error handling here? To make the test more robust? Check out https://jestjs.io/docs/en/asynchronous to see how they were handling errors with asynchronous tests.

File: test/test.js [hunk 3]
```diff
@@ -36,4 +37,96 @@ describe('Database interactions', () => {
...
+      });
+      id += 1;
+    }
+  });
+  test('Should get ratings for all product id entries & should be between 1-5', (done) => {
+    let productId = 0;
+    const queryString = 'SELECT * FROM user_reviews WHERE url_id = ?';
+    const promises = [];
+    while (productId < 100) {
+      expect.assertions(200);
```
Comment (andrewdeanyoder) [inline test/test.js]: I'm curious. Why do you need the expect.assertions again? I would expect it to fail the first 99 times the while loop completes. I'm assuming it doesn't, but it seems to be unnecessary. If you do need it, wouldn't it be cleaner to have it outside the while loop? Or, you could put it inside the if(promises.length === 100) block.

**Review by andrewdeanyoder — APPROVED** [review]
@Matt-Wigg Overall I think it looks good. I suggested two changes that would improve things, but I'm going to select "approve" here so you can continue to make progress on the front-end and make those changes whenever you have time.

Nice work pulling this together on limited time. I'm definitely going to use a couple of your methods in my tests.

File: test/test.js (hunk 2 above)
Comment (matt-wigg) [reply test/test.js:47]: This is great. Thank you!!! I've implemented the error handling by checking it err exists and passing It to done() if it does! - working as intended 🐮

File: test/test.js (hunk 3 above)
Comment (matt-wigg) [reply test/test.js]: Great suggestion: inside the if statement makes much more sense as I can catch any errors up to that point. I'll make this change!

**Pull request merged**